### PR TITLE
Add support for disk in Python job client

### PR DESCRIPTION
**Pull request by kathryn-zhou**
## Changes proposed in this PR
- Add optional Disk class for Python jobs
- Add unit tests for different combinations of specifying disk

## Why are we making these changes?
- Allow Python job client users to specify disk request, limit, and type

### CODE REVIEW

File: jobclient/python/tests/test_job.py [hunk 1]
```diff
@@ -157,7 +157,7 @@
     'progress_output_file': 'output.txt',
     'progress_regex_string': 'test',
     'gpus': 1,
-    'ports': 10
+    'ports': 10,
```
Comment (dposada) [inline jobclient/python/tests/test_job.py:160]: Why the added `,`?

File: jobclient/python/tests/test_job.py [hunk 2]
```diff
@@ -185,6 +185,27 @@
     ]
 }}
 
+JOB_DICT_DISK_ONLY_REQUEST = {**JOB_DICT_NO_OPTIONALS, **{
+    'disk': {
+        'request': 10.0,
```
Comment (dposada) [inline jobclient/python/tests/test_job.py:190]:
```suggestion
        'request': 10.0
```

File: jobclient/python/tests/test_job.py [hunk 3]
```diff
@@ -286,9 +308,22 @@ def _check_group(self, job: Job, jobdict: dict):
         else:
             self.assertIsNone(job.group)
 
+    def _check_disk(self, job: Job, jobdict: dict):
+        """Check the disk field of a Job.
+
+        The Job's disk attribute should be either:
+        * The value of `jobdict['disk']` if set,
+        * Otherwise, the value of `jobdict['disk'][0]` if set,
```
Comment (dposada) [inline jobclient/python/tests/test_job.py:316]: I don't understand this bullet

File: jobclient/python/tests/test_job.py (hunk 1 above)
Comment (kathryn-zhou) [reply jobclient/python/tests/test_job.py:160]: ah I was originally adding disk to this job dict but then I removed it to make a more exhaustive test elsewhere. removed the `,`

File: jobclient/python/tests/test_job.py (hunk 2 above)
Comment (kathryn-zhou) [reply jobclient/python/tests/test_job.py:190]: done

File: jobclient/python/tests/test_job.py (hunk 3 above)
Comment (kathryn-zhou) [reply jobclient/python/tests/test_job.py:316]: forgot to update docstring, that bullet is irrelevant and removed now

**Comment by kathryn-zhou** [comment]
Received approval on Slack from Cindy Ge on Modeling Tools.

**Review by dposada — APPROVED** [review]

**Pull request merged**